docs/source/examples/e2e_flow.rst

```diff
@@ -0,0 +1,438 @@
+        # match what's shown in the logs above
+        checkpoint_files: [
+            hf_model_0001_0.pt,
+            hf_model_0002_0.pt,
+        ]
+
+        output_dir: <checkpoint_dir>
+        model_type: LLAMA2
+
+
+Once the config is updated, let's kick off quantization! We'll use the default
+quantization method from the config.
+
+
+.. code-block:: bash
+
+    tune run quantize --config quantize
+
+Once quantization is complete, you'll see the following in the logs.
+
+.. code-block:: bash
+
+    [quantize.py:68] Time for quantization: 19.76 sec
+    [quantize.py:69] Memory used: 13.95 GB
+    [quantize.py:82] Model checkpoint of size 3.67 GB saved to <checkpoint_dir>/hf_model_0001_0-4w.pt
+
+
+.. note::
+    Unlike the fine-tuned checkpoints, this output a single checkpoint file. This is
+    because our quantization APIs currently don't support any conversion across formats.
+    As a result you won't be able to use these quantized models outside of TorchTune.
+    But you should be able to use these with the generation and evaluation recipes within
+    TorchTune. These results will help inform which quantization methods you should use
+    with your favorite inference engine.
+
+Now that we have the quantized model. Let's rerun generation.
+
+Let's modify ``custom_generation_config.yaml`` to include the following changes.
+
+.. code-block:: yaml
+
+    checkpointer:
+        # we need to use the custom TorchTune checkpointer
+        # instead of the HF checkpointer for loading
+        # quantized models
+        _component_: torchtune.utils.FullModelTorchTuneCheckpointer
+
+        # directory with the checkpoint files
+        # this should match the output_dir specified during
+        # finetuning
+        checkpoint_dir: <checkpoint_dir>
+
+        # checkpoint files point to the quantized model
+        checkpoint_files: [
+            hf_model_0001_0-4w.pt,
+        ]
+
+        output_dir: <checkpoint_dir>
+        model_type: LLAMA2
+
+    # we also need to update the quantizer to what was used during
+    # quantization
+    quantizer:
+        _component_: torchtune.utils.quantization.Int4WeightOnlyQuantizer
+        groupsize: 256
+
+
+Once the config is updated, let's kick off generation! We'll use the
+same sampling parameters as before.
+
+We'll use a different prompt from the one in the config
+
+.. code-block:: bash
+
+    tune run generate \
+    --config generate \
+    prompt="What are some interesting sites to visit in th Bay Area?"
+
+
+Once generation is complete, you'll see the following in the logs.
+
+
+.. code-block:: bash
+
+    [generate.py:92] A park in San Francisco that sits at the top of a big hill.
+                     There are lots of trees and a beautiful view of San Francisco...
+
+    [generate.py:96] Time for inference: 4.13 sec total, 72.62 tokens/sec
+    [generate.py:99] Memory used: 17.85 GB
+
+With quantization (and torch compile under the hood), we've sped up generation
```

Comment:
The mention of compile here feels weird, should explicitly call out how we're using it prior to the last sentence

Reply:
I dont want to spend too much time on this tbh

Reply:
OK fair enough. Mainly it's unclear how we are actually using it, like is it a core part of the generation recipe or no? Nbd either way though